### PR TITLE
feat: [PIE-4689]: execution list view menu

### PR DESCRIPTION
**Pull request by udayvunnam**
### Summary

- execution list view menu actions update, show only in details view and menu style
- primary navigation link goes to execution details

#### Screenshots

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 16 Issues 
* Code issue 16 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11229)

File: src/modules/70-pipeline/pages/pipeline-list/PipelineListTable/PipelineListCells.tsx [hunk 1]
```diff
@@ -274,16 +273,9 @@ export const MenuCell: CellType = ({ row, column }) => {
 
   return (
     <Layout.Horizontal style={{ justifyContent: 'flex-end' }}>
-      <Popover
-        isOpen={menuOpen}
-        onInteraction={nextOpenState => {
-          setMenuOpen(nextOpenState)
-        }}
```
Comment (udayvunnam) [inline src/modules/70-pipeline/pages/pipeline-list/PipelineListTable/PipelineListCells.tsx:281]: additional state management for popover unnecessary

File: src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionStage.tsx [hunk 2]
```diff
@@ -29,8 +29,8 @@ export interface ExecutionStageProps {
 }
 
 const stageIconMap: Partial<Record<StageType, IconName>> = {
-  [StageType.BUILD]: 'cd-main',
-  [StageType.DEPLOY]: 'ci-main',
+  [StageType.BUILD]: 'ci-main',
+  [StageType.DEPLOY]: 'cd-main',
```
Comment (udayvunnam) [inline src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionStage.tsx:33]: bug fix

File: src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx [hunk 3]
```diff
@@ -126,31 +114,24 @@ export const PipelineNameCell: CellType = ({ row }) => {
...
-            className={css.tags}
-            tags={defaultTo(data?.tags, []).reduce((_tags, tag) => {
-              _tags[tag.key] = tag.value
-              return _tags
-            }, {} as { [key: string]: string })}
-          />
-        )}
-      </Layout.Horizontal>
+    <Layout.Horizontal spacing="small" style={{ alignItems: 'center' }}>
       <Link to={toExecutionPipelineView}>
```
Comment (udayvunnam) [inline src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx:118]: link to go to execution details

File: src/modules/70-pipeline/components/ExecutionActions/ExecutionActions.tsx [hunk 4]
```diff
@@ -302,112 +321,108 @@ export default function ExecutionActions(props: ExecutionActionsProps): React.Re
...
                 onClick={reRunPipeline}
               />
             )}
-            <MenuItem text={getString(pauseText)} onClick={pausePipeline} disabled={!canPause} />
             <MenuItem text={getString(abortText)} onClick={openAbortDialog} disabled={!canAbort} />
+            <MenuItem text={getString(pauseText)} onClick={pausePipeline} disabled={!canPause} />
             <MenuItem text={getString(resumeText)} onClick={resumePipeline} disabled={!canResume} />
-            {onViewCompiledYaml && (
-              <MenuItem text={getString('pipeline.execution.actions.viewCompiledYaml')} onClick={onViewCompiledYaml} />
+            {!isExecutionDetailsView && (
```
Comment (udayvunnam) [inline src/modules/70-pipeline/components/ExecutionActions/ExecutionActions.tsx:406]: fix: these links shouldn't be shown in execution details view

**Review by satyam-harness — DISMISSED** [review]

**Review by vardanbansal-harness — DISMISSED** [review]

**Review by satyam-harness — APPROVED** [review]

**Review by vardanbansal-harness — APPROVED** [review]

**Pull request merged**